docs/apps/python.md

```diff
@@ -5,256 +5,97 @@ tags:
-Desk](../support/contact.md).
+## Usage
 
+It is generally recommended to use one of the
+[pre-installed Python environments](python.md#pre-installed-python-environments),
+since these already contain the essential libraries for most uses.
+If for some reason one wishes to use Python without loading an environment
+module, a
+basic
+[system Python](python.md#system-python)
+is installed by default.
 
-### Using `venv`
+See our
+[Python usage guide](../support/tutorials/python-usage-guide.md)
+for instructions on using Python effectively on CSC supercomputers.
 
-The recommended way to add packages on top of an existing environment
-is to use [venv](https://docs.python.org/3/tutorial/venv.html), which
-is a standard Python module for creating a lightweight "virtual
-environment". You can have multiple virtual environments, for example
-one for each project.
+### System Python
 
-For example to install a package called `whatshap` on top of the
-CSC-provided [python-data](python-data.md) module:
+If using a pre-installed environment is not suitable,
+the basic system Python 3.9 can be launched with:
 
 ```bash
-cd /projappl/<your_project>  # change this to the appropriate path for your project
-module load python-data
-python3 -m venv --system-site-packages venv
-source venv/bin/activate
-pip install whatshap
+python3.9
 ```
 
 !!! warning
-
-    Don't forget to use the `--system-site-packages` flags when creating
-    the virtual environment, otherwise the environment will not find the
-    pre-installed packages from the base module (for example numpy from
-    python-data).
-
-
-Later when you wish to use the virtual environment you only need to
-load the module and activate the environment:
+	It is strongly advised to explicitly launch Python version 3.9 as
+	above, since **the default version launched by `python3` (3.6.8) has
+	[reached end-of-life](https://devguide.python.org/versions/)**.
+
+### Pre-installed Python environments
+
+Puhti and Mahti have several pre-installed
+[environment modules](../computing/modules.md) containing
+Python environments made for different science areas.
+For more details about the Python versions and libraries that are available
+for a module, please see the corresponding application page by opening
+one of the links in the table below.
+
+| Module name | Purpose | Package list |
+|-|-|-|
+| [biopythontools](biopython.md) | bioinformatics | [pip](https://a3s.fi/python-pkg-lists/biopythontools_3.10.6.txt) |
+| [geoconda](geoconda.md) | geoinformatics | [conda](https://a3s.fi/python-pkg-lists/geoconda_3.10.9.yml) |
+| [jax](jax.md) | JAX ML framework | [pip](https://a3s.fi/python-pkg-lists/jax0.4.23_python3.9_cuda12.2_csc_fix1.txt) |
+| [python-data](python-data.md) | data analysis and ML utilities | [conda](https://a3s.fi/python-pkg-lists/python-data-2023-11.yaml) |
+| [pytorch](pytorch.md) | PyTorch ML framework | [pip](https://a3s.fi/python-pkg-lists/pytorch_2.2.1_csc_fix2.txt) |
```

Comment:
How are these lists generated? And is it automatic? At least for PyTorch it looks like it might just be the requirements file I have used in installation. This is a bit problematic as I use the requirements file only for "simple" installations. It doesn't include packages "manually" installed in the singularity recipe, like for example torch and deepspeed.

Reply:
I generated new ones using `list-packages` for the container-based environments and `pip list` for `biopythontools`. I wrote a little script for doing that, maybe it could be set up to update them once a month or so?